Bugfix: QueryCreationException in ProductRepository

`Caused by: org.springframework.data.repository.query.QueryCreationException: Could not create query for public abstract boolean com.lutzarDemos.shoppingdemo.repository.ProductRepository.existByNameAndBrand(java.lang.String,java.lang.String); Reason: Failed to create query for method public abstract boolean com.lutzarDemos.shoppingdemo.repository.ProductRepository.existByNameAndBrand(java.lang.String,java.lang.String); No property 'existByName' found for type 'Product'`

--- a/src/main/java/com/lutzarDemos/shoppingdemo/controller/ProductController.java
+++ b/src/main/java/com/lutzarDemos/shoppingdemo/controller/ProductController.java
@@ -65,8 +65,8 @@ public class ProductController {
             Product theProduct = productService.addProduct(product);
             return ResponseEntity.ok(new ApiResponse("Add product success!", theProduct));
 
-        } catch (AlreadyExistsException e) {
-            return ResponseEntity.status(CONFLICT).body(new ApiResponse(e.getMessage(), null));
+        } catch (Exception e) {
+            return ResponseEntity.status(INTERNAL_SERVER_ERROR).body(new ApiResponse(e.getMessage(), null));
         }
     }
 

--- a/src/main/java/com/lutzarDemos/shoppingdemo/repository/ProductRepository.java
+++ b/src/main/java/com/lutzarDemos/shoppingdemo/repository/ProductRepository.java
@@ -18,5 +18,5 @@ public interface ProductRepository extends JpaRepository<Product, Long> {
     List<Product> findByName(String name);
     List<Product> findByBrandAndName(String brand, String name);
     Long countByBrandAndName(String brand, String name);
-    boolean existByNameAndBrand(String name, String brand);
+    //boolean existByNameAndBrand(String name, String brand);
 }

--- a/src/main/java/com/lutzarDemos/shoppingdemo/service/product/ProductService.java
+++ b/src/main/java/com/lutzarDemos/shoppingdemo/service/product/ProductService.java
@@ -48,11 +48,12 @@ public class ProductService implements IProductService{
      */
     @Override
     public Product addProduct(AddProductRequest request) {
+        /*
         if (productExists(request.getName(), request.getBrand())) {
             throw new AlreadyExistsException
                     (request.getBrand() + " " + request.getName() + " Already Exists! Try Updating The PRODUCT instead!");
         }
-
+        */
         Category category = Optional
                 .ofNullable(categoryRepository.findByName(request.getCategory().getName()))
                 .orElseGet(()-> {
@@ -69,10 +70,11 @@ public class ProductService implements IProductService{
      * @param name      Name of the PRODUCT
      * @param brand     Brand of the PRODUCT
      */
+    /*
     private boolean productExists(String name, String brand) {
         return productRepository.existByNameAndBrand(name, brand);
     }
-
+    */
     /**
      * Helper method to create a new PRODUCT
      *

